controllers/pod_node_module_reconciler.go

```diff
@@ -77,7 +77,7 @@ func (pnmr *PodNodeModuleReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		if !pod.DeletionTimestamp.IsZero() {
 			logger.Info("Pod deletion requested; removing finalizer")
 
-			if err := pnmr.deleteFinalizer(ctx, &pod); err != nil {
+			if err := pnmr.deleteFinalizer(ctx, &pod); err != nil && !k8serrors.IsNotFound(err) {
```

Comment:
Great catch! Can you just add a comment explaining this race?

Reply:
Adding a comment explaining this case makes absolute sense :+1:  Please let me know if what I added [here](https://github.com/kubernetes-sigs/kernel-module-management/pull/251/files#diff-80d58b0351ebaad9c427b40c246f866df41ecb8079741e35005d66ead9af490aR80-R83) is sufficient, otherwise I can try and rephrase it.

Reply:
Would [`IgnoreNotFound`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#IgnoreNotFound) be fancier here?
```suggestion
			if err := pnmr.deleteFinalizer(ctx, &pod); client.IgnoreNotFound(err) != nil {
```